packages/glob/src/internal-pattern-helper.ts

```diff
@@ -0,0 +1,142 @@
+import * as pathHelper from './internal-path-helper'
+import {IGlobOptions} from './internal-glob-options'
+import {MatchResult} from './internal-match-result'
+import {Pattern} from './internal-pattern'
+
+const IS_WINDOWS = process.platform === 'win32'
+
+/**
+ * Given an array of patterns, returns an array of paths to search.
+ * Duplicates and paths under other included paths are filtered out.
+ */
+export function getSearchPaths(patterns: Pattern[]): string[] {
+  // Ignore comment and negate patterns
+  patterns = patterns.filter(x => !x.comment && !x.negate)
+
+  // Create a map of all search paths
+  const searchPathMap: {[key: string]: string} = {}
+  for (const pattern of patterns) {
+    const key = IS_WINDOWS
+      ? pattern.searchPath.toUpperCase()
```

Comment:
Is this to ensure we get case insensitive matching on Windows?

Reply:
yes

Reply:
note case is preserved for the search path, this just helps filter out unnecessary search paths on windows (ones that exist under a higher search path)